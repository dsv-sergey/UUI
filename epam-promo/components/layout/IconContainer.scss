.root {
    fill: var(--fill-color);
}