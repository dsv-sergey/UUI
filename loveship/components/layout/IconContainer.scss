@use '../../assets/styles/scss/index'  as *;

.root {
    fill: var(--uui-color-700);
}
