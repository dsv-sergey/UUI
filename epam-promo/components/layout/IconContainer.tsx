import { IconContainer as uuiIconContainer, ControlIconProps } from '@epam/uui-components';
import { withMods } from '@epam/uui';
import { ColorMod } from '../types';
import * as css from './IconContainer.scss';
import * as styles from '../../assets/styles/colorvars/layout/iconContainer-colorvars.scss';

export interface IconContainerMods extends ColorMod {
}

export function applyIconContainerMods(mods: IconContainerMods) {
    return [
        css.root,
        styles['icon-container-color-' + (mods.color || 'gray60')],
    ];
}

export const IconContainer = withMods<ControlIconProps, IconContainerMods>(
    uuiIconContainer,
    applyIconContainerMods,
);