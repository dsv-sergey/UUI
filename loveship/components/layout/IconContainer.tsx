import { IconContainer as uuiIconContainer, ControlIconProps } from '@epam/uui-components';
import { withMods } from '@epam/uui';
import { ColorMod } from "../types";
import * as css from './IconContainer.scss';
import * as styles from '../../assets/styles/scss/loveship-color-vars.scss';

export interface IconContainerMods extends ColorMod {
}

export function applyIconContainerMods(mods: IconContainerMods) {
    return [
        css.root,
        styles['color-' + (mods.color || 'night600')],
    ];
}

export const IconContainer = withMods<ControlIconProps, IconContainerMods>(
    uuiIconContainer,
    applyIconContainerMods,
);