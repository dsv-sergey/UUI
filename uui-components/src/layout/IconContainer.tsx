import * as React from 'react';
import cx from 'classnames';
import * as css from './IconContainer.scss';
import { uuiElement, uuiMod, uuiMarkers, IHasCX, IDisableable, Icon } from '@epam/uui';
import { Svg } from '../widgets';

export interface ControlIconProps extends IHasCX, IDisableable {
    icon?: Icon;
    flipY?: boolean;
    rotate?: '0' | '90cw' | '180' | '90ccw';
    onClick?(e: React.SyntheticEvent<HTMLDivElement>): void;
    tabIndex?: number;
    style?: React.CSSProperties;
    size?: number;
}

export const IconContainer = (props: ControlIconProps) => {
    const isClickable = !props.isDisabled && props.onClick;

    return (
        <div
            className={ cx(
                css.container,
                uuiElement.icon,
                props.isDisabled ? uuiMod.disabled : uuiMod.enabled,
                isClickable && uuiMarkers.clickable,
                props.cx,
            ) }
            onClick={ isClickable ? props.onClick : undefined }
            tabIndex={ props.tabIndex }
            style={ props.style }
        >
            <Svg svg={ props.icon } width={ props.size } height={ props.size } cx={ cx(props.flipY && css.flipY, props.rotate && css['rotate-' + props.rotate]) }/>
        </div>
    );
};