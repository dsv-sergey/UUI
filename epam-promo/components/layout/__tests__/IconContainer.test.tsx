import React from 'react';
import renderer from 'react-test-renderer';
import { IconContainer } from '../IconContainer';
import * as calendarIcon from '../../../icons/calendar-18.svg';

describe('IconContainer', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<IconContainer
                icon={ calendarIcon }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<IconContainer
                color='blue'
                onClick={ () => null }
                icon={ calendarIcon }
                size={ 36 }
                rotate='90ccw'
                tabIndex={ 0 }
                flipY
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});