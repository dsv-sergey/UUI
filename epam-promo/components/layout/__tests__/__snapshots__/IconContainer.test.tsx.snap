// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconContainer should be rendered correctly 1`] = `
<div
  className="container uui-icon uui-enabled root icon-container-color-gray60"
>
  <svg
    className=""
    height={24}
    viewBox="0 0 12 24"
    width={12}
  >
    <use
      xlinkHref="#calendar-18.svg"
    />
  </svg>
</div>
`;

exports[`IconContainer should be rendered correctly with props 1`] = `
<div
  className="container uui-icon uui-enabled -clickable root icon-container-color-blue"
  onClick={[Function]}
  tabIndex={0}
>
  <svg
    className="flipY rotate-90ccw"
    height={36}
    viewBox="0 0 12 24"
    width={36}
  >
    <use
      xlinkHref="#calendar-18.svg"
    />
  </svg>
</div>
`;
